awx/main/tests/functional/test_rbac_organization.py

```diff
@@ -36,3 +36,15 @@ def test_organization_access_user(cl, organization, user):
     org = access.get_queryset()[0]
     assert len(org.admin_role.members.all()) == 0
     assert len(org.member_role.members.all()) == 1
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('ext_auth', [True, False])
+def test_org_resource_role(ext_auth, organization, rando, org_admin):
+    with mock.patch('awx.main.access.settings') as settings_mock:
+        settings_mock.MANAGE_ORGANIZATION_AUTH = ext_auth
+        access = OrganizationAccess(org_admin)
+
+        assert access.can_attach(organization, rando, 'member_role.members') == ext_auth
```

Comment:
Speaking for the case of `ext_auth=False`

Why should `org_admin` not be able to add `rando` as a member? If `rando` is a member of a different organization, then I agree, but I don't think that's the case for the rando fixture by default. We have a provision to allow org admins add users to their org if that user is not superuser (obviously) and orphaned from all other organizations. This is to support to flow of (1) org admin creates new user by POST to /api/v2/users/ (2) subsequently adds that user as a member of their organization. Otherwise they have no choice but to use /api/v2/organizations/N/users/, which is a really nasty entanglement of REST principles with RBAC restrictions.

Reply:
By my understanding of the conversation with Bill, under `settings.MANAGE_ORGANIZATION_AUTH = False` organization and team membership may not be modified by anyone short of a system superuser.  Since this scenario is meant to support external management via LDAP, I don't believe that user creation is allowed either.

Reply:
https://github.com/ansible/awx/blob/1ddb675fa2e89ae3de6e37a857257d5384c838bf/awx/main/access.py#L574-L583

Reply:
oh man, I got the default backwards. Sorry about.

So this test is asserting that `org_admin` _can_ add `rando` as a member under normal circumstances. That sounds fine.

I would also want to get coverage of the scenario where this action cannot be done, due to some non-trivial reason expressed in `RoleAccess`, because that's the change being made, `OrganizationAccess` is deferring to the role access class.

Reply:
We already have coverage for the scenario I mentioned, but we don't have coverage in the call pattern that you employ here (and that is the thing you are fixing). One instance of the existing coverage is:

https://github.com/ansible/awx/blob/87d55b13bcd88d46f9ca13c937177062bc5607d8/awx/main/tests/functional/test_rbac_role.py#L162-L168

(I don't know how far this rabbit hole goes, I just picked the first ones I saw)

Where this does `assert not role_access.can_attach(org1.admin_role, org12_member, 'members', None)`, it should also do `assert not org_access.can_attach(org1, org12_member, 'admin_role.members', None) `, assuming `org_access` is set up with similar patterns.

That is a scenario that we should have coverage for.

Reply:
Done.